verification/cbmc/proofs/aws_cryptosdk_priv_try_gen_key/aws_cryptosdk_priv_try_gen_key_harness.c

```diff
@@ -0,0 +1,244 @@
+int aws_cryptosdk_hkdf(
+    struct aws_byte_buf *okm,
+    enum aws_cryptosdk_sha_version which_sha,
+    const struct aws_byte_buf *salt,
+    const struct aws_byte_buf *ikm,
+    const struct aws_byte_buf *info) {
+    assert(aws_byte_buf_is_valid(salt));
+    assert(aws_byte_buf_is_valid(ikm));
+    assert(aws_byte_buf_is_valid(info));
+    if (nondet_bool()) {
+        return AWS_OP_ERR;
+    }
+    return AWS_OP_SUCCESS;
+}
+
+/* Stub this because of the override of aws_array_list_get_at_ptr and for performance.
+ The contents of session->keyring_trace are nondet in the construction of the harness.
+ Also neither keyring trace is later used.
+ Original function is here:
+ https://github.com/aws/aws-encryption-sdk-c/blob/master/source/list_utils.c#L17 */
+int aws_cryptosdk_transfer_list(struct aws_array_list *dest, struct aws_array_list *src) {
+    assert(src != dest);
+    assert(aws_array_list_is_valid(dest));
+    assert(aws_array_list_is_valid(src));
+    assert(dest->item_size == src->item_size);
+
+    if (nondet_bool()) {
+        return AWS_OP_ERR;
+    }
+    return AWS_OP_SUCCESS;
+}
+
+void array_list_item_generator(struct aws_array_list *elems) {
+    assert(elems->item_size == sizeof(struct aws_hash_element));
+    for (size_t index = 0; index < elems->length; ++index) {
+        struct aws_hash_element *val = (struct aws_hash_element *)((uint8_t *)elems->data + (elems->item_size * index));
+        /* Due to the cast to uint16, the entire size of the enc_ctx must be less than < UINT16_MAX
+         * This is a simple way to ensure this without a call to enc_ctx_size. */
+        struct aws_string *key = ensure_string_is_allocated_nondet_length();
+        __CPROVER_assume(aws_string_is_valid(key));
+        __CPROVER_assume(key->len <= UINT8_MAX);
+        val->key                 = key;
+        struct aws_string *value = ensure_string_is_allocated_nondet_length();
+        __CPROVER_assume(aws_string_is_valid(value));
+        __CPROVER_assume(value->len <= UINT8_MAX);
+        val->value = value;
+    }
+}
+
+/* Stub this because of the override of aws_array_list_get_at_ptr
+ Original function is here:
+ https://github.com/aws/aws-encryption-sdk-c/blob/master/source/keyring_trace.c#L235 */
+void aws_cryptosdk_keyring_trace_clear(struct aws_array_list *trace) {
+    AWS_FATAL_PRECONDITION(aws_cryptosdk_keyring_trace_is_valid(trace));
+    AWS_FATAL_PRECONDITION(trace->item_size == sizeof(struct aws_cryptosdk_keyring_trace_record));
+    aws_array_list_clear(trace);
+}
+
+/* Stub this until https://github.com/diffblue/cbmc/issues/5344 is fixed
+ Original function is here:
+ https://github.com/aws/aws-encryption-sdk-c/blob/master/source/edk.c#L44 */
+void aws_cryptosdk_edk_list_clean_up(struct aws_array_list *encrypted_data_keys) {
+    assert(aws_cryptosdk_edk_list_is_valid(encrypted_data_keys));
+    aws_array_list_clean_up(encrypted_data_keys);
+}
```

Comment:
We might need all these stubs in other proofs. Why are we adding them here and not in the stub folder? I'd advocate to move them there.